p4-16/spec/P4-16-spec.adoc

```diff
@@ -9547,35 +9547,35 @@ The next table lists restrictions on what kinds of calls can be made
 from which places in a P4 program.  Calling a parser, control, or
 table means invoking its `apply()` method. Calling a value-set means using it
 in a select expression. The row for `extern` describes where extern method
-calls can be made from.
+or function calls can be made from.
 
 One way that an extern can be called from the top level of a parser or
 control is in an initializer expression for a declared variable,
 e.g. `bit<32> x = rand.get();`.
 
-[.center,cols="3,^2,^2,^2,^2,^2,^2",width=90%]
+[.center,cols="3,^2,^2,^2,^2,^2,^2,^2",width=90%]
 |===
-| This type 6+^| can be called at run time from this place in a P4 program
+| This type 7+^| can be called at run time from this place in a P4 program
 
-|             .>| parser state .>| control apply block .>| parser or top level .>| action .>| extern .>| function
+|             .>| top level .>| parser state .>| control apply block .>| top level within a parser or control .>| action .>| extern .>| function
```

Comment:
@rcgoodfellow Regarding your question during the meeting about the meaning of "top level" and "top level within a parser or control", if the text above the table said the following, would that make it clear for you?

"Top level" refers to the part of a P4 program that one is in at the beginning of the file, where all declarations have global scope, i.e. not within any control, parser, or function definition.

"Top level within or control" means either within a parser definition, but outside of any of its parser state definitions, or within a control definition, but outside of its `apply { ... }` block.

Reply:
SGTM, thanks!